File: tournaments/countTriangles/countTriangles.js
function countTriangles(x, y) {
    let result = 0;
    
    for (let i = 0; i in x; ++i) {
        for (let j = i + 1; j in x; ++j) {
            for (let k = j + 1; k in x; ++k) {
                if ((x[i] - x[j])*(y[i] - y[k]) !== (x[i] - x[k])*(y[i] - y[j])) {
                    ++result;
                }
            }
        }
    }
    
    return result;
}

File: tournaments/countTriangles/countTriangles.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub countTriangles {
    my ($x, $y) = @_;

    my $n = @$x;
    my $result = 0;

    for (my $i = 0; $n > $i; ++$i) {
        for (my $j = $i + 1; $n > $j; ++$j) {
            for (my $k = $j + 1; $n > $k; ++$k) {
                if ((@$x[$i] - @$x[$j])*(@$y[$i] - @$y[$k]) != (@$x[$i] - @$x[$k])*(@$y[$i] - @$y[$j])) {
                    ++$result;
                }
            }
        }
    }

    return $result;
}
